### PR TITLE
Adding a solution for max-of-vector 

### PR DESCRIPTION
**Pull request by MoreThanADC**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: max-of-vector
🏅 Granted 5 XP for @MoreThanADC

**Comment by ziobron** [comment]
⏱ 3 XP extra for @MoreThanADC for on-time delivery

File: homework/max-of-vector/maxOfVector.hpp [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
+/*
+Please review this code. Which solution is better?
+
+int maxOfVector(const std::vector<int>& vec) {
+    std::vector<int> newVector = vec;
+    std::sort(newVector.begin(), newVector.end());
+    return newVector.at(newVector.size()-1);
+}
+
+*/
```
Comment (mariuszlisowski) [inline homework/max-of-vector/maxOfVector.hpp:25]: Newline at the end of a file needed

File: homework/max-of-vector/maxOfVector.hpp [hunk 2]
```diff
@@ -1,8 +1,25 @@
 #pragma once
 #include <vector>
 #include <limits>
```
Comment (mariuszlisowski) [inline homework/max-of-vector/maxOfVector.hpp:3]: Header not used

File: homework/max-of-vector/maxOfVector.hpp [hunk 3]
```diff
@@ -1,8 +1,25 @@
...
+#include <algorithm>
+
 
 int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
+    int max = vec.at(0);
```
Comment (nauka-programowania-MA) [inline homework/max-of-vector/maxOfVector.hpp:8]: Should be const.

File: homework/max-of-vector/maxOfVector.hpp [hunk 4]
```diff
@@ -1,8 +1,25 @@
...
+    for_each(vec.begin(), vec.end(), [&max](int element){
+        if (element > max) {
+            max = element;
+        }
+    });
```
Comment (nauka-programowania-MA) [inline homework/max-of-vector/maxOfVector.hpp:13]: If you want to use algorithm function, please use dedicated one. Here you should use std::max_element.

File: homework/max-of-vector/maxOfVector.hpp [hunk 5]
```diff
@@ -1,8 +1,25 @@
...
+    return max;
 }
+/*
+Please review this code. Which solution is better?
+
+int maxOfVector(const std::vector<int>& vec) {
+    std::vector<int> newVector = vec;
+    std::sort(newVector.begin(), newVector.end());
+    return newVector.at(newVector.size()-1);
```
Comment (nauka-programowania-MA) [inline homework/max-of-vector/maxOfVector.hpp:22]: This is `O(log(n))` so worse case then use `std::max_element` which is `O(n)`.

File: homework/max-of-vector/maxOfVector.hpp (hunk 1 above)
Comment (nauka-programowania-MA) [reply homework/max-of-vector/maxOfVector.hpp:25]: Actually we don't need it in modern compiler. This is legacy form C.

**Review by nauka-programowania-MA — APPROVED** [review]

File: homework/max-of-vector/maxOfVector.hpp (hunk 3 above)
Comment (nauka-programowania-MA) [inline homework/max-of-vector/maxOfVector.hpp:8]: mark as const

**Pull request closed without merging**